core/src/test/java/org/apache/calcite/avatica/util/DateAccessorTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.avatica.util;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Date;
+import java.sql.SQLException;
+import java.util.Calendar;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class DateAccessorTest {
+
+  private static final Calendar UTC =
+      Calendar.getInstance(TimeZone.getTimeZone("UTC"), Locale.ROOT);
+
+  private Cursor.Accessor instance;
+  private Calendar localCalendar;
+  private Date value;
```

Comment:
I think it's OK. If a test class has 5 test methods, then junit will create 5 instances of the class. Whether the tests are run in parallel or series, the instance variables won't conflict.

The main reason to set variables in \@Before and clear them in \@After is to prevent excessive memory leaks (and to free resources such as DB connections after the test has finished).

My personal style is to put everything on the stack of the test method, because I am suspicious of non-final fields. But I don't expect everyone to follow my style.

Reply:
Thanks @julianhyde for the explanation, since it's safe I leave it up to @freastro to decide what's best here

Reply:
Thanks for the suggestion. I like to push common test code to the `@Before` method. I think it makes the tests more readable.